core_lib/src/movieexporter.cpp

```diff
@@ -479,12 +485,10 @@ Status MovieExporter::generateMovie(
 
             obj->paintImage(painter, currentFrame, false, true);
 
-            QByteArray imgData;
-            QBuffer buffer(&imgData);
-            // BMPs are used because QT encodes them faster than other formats (less compression)
-            bool bSave = imageToExport.save(&buffer, "BMP", 100);
-            Q_ASSERT(bSave);
-            ffmpeg.write(imgData);
+            // Should use sizeInBytes instead of byteCount to support large images,
+            // but this is only supported in QT 5.10+
+            bytesWritten = ffmpeg.write(reinterpret_cast<const char*>(imageToExport.constBits()), imageToExport.byteCount());
+            Q_ASSERT(bytesWritten == imageToExport.byteCount());
```

Comment:
According to docs, QIODevice::write can return -1 when an error occurs. I suppose if that happened (maybe when FFmpeg dies while being written to or something?) it would trigger the assertion and potentially crash the program. I believe it would be better to account for such potential errors.

Reply:
How exactly should it be handled? If it fails, retrying would be fairly pointless as it's not likely to suddenly work the second time. Throwing an error to the user would be ideal, but at the moment there is no error handling/dialog for any export errors, so to handle all errors would require more work beyond these lines. With the asserts, at least we know that part is failing. During releases, it will not crash the program. Adding full error handling to the export is on my todo list, but I think could be left for separate PR later.

Reply:
Hmm, fair points. Maybe leave a TODO though or make an issue to avoid forgetting about it.